pages/locationdetail/[id].js

```diff
@@ -55,10 +65,32 @@ export default function LocationDetailsPage({
     setColonies(colonies.filter((colony) => colony.id !== colonyToDelete.id));
   }
 
+  function handleInputChange(event) {
+    const { name, value } = event.target;
+    setFormData((prevFormData) => ({ ...prevFormData, [name]: value }));
+  }
+
   return (
     <>
       <StyledSection>
         <h2>{currentLocation?.locationName}</h2>
+
+        <StyledQuestionCard padding="0.5rem 0" width="70%" margin="0 0 2rem 0">
+          <StyledQuestionLabel htmlFor="generalNotesLocation">
+            General notes about the location:
+          </StyledQuestionLabel>
+          <StyledQuestionTextArea
+            id="generalNotesLocation"
+            name="generalNotesLocation"
+            type="text"
+            rows="3"
+            maxLength="100"
+            minLength="3"
+            value={formData.generalNotesLocation || ""}
+            onChange={handleInputChange}
+          />
+        </StyledQuestionCard>
```

Comment:
since you have this exact code twice please put it into a component (e.g. `ColonyNotesForm`) that abstracts this form including the state, the handle input function etc..